### PR TITLE
NHSO-43209: Update App page Enums in JS Api Specification

### PR DESCRIPTION
**Pull request by sanjaykumarNHS24**
Update App page Enums in JS Api Specification

### CODE REVIEW

File: src/js-v2-api-specification.md [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+`nhsapp.navigation.clearBackAction()`
+
+##### Description
+
+Clear the action that will be executed when a user executes the native back functionality on their device such as pressing the back button on Android or swiping back on iOS.
+        
+
+##### Example Usage
+
+```js
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md]: Consistency with above: ```javascript

File: src/js-v2-api-specification.md [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+##### Parameters
+
+| Parameters              |                                                                                                                                                                |
+|-------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| backAction                 | **Type:** Function *Required*<br>The function to be excuted when the native back is pressed.            
+
+##### Example Usage
+
+```js
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md]: Consistency with above: ```javascript

File: src/js-v2-api-specification.md [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+| HOME_PAGE      | NHS App Home Page      |
+| SERVICES       | Services Hub Page      |
+| YOUR_HEALTH    | Your Health Hub Page   |
+| MESSAGES       | Messages Hub Page      |
+| UPLIFT         | P9 Uplift Page         |
+
+
+##### Example Usage
+
+```js
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md]: Consistency with above: ```javascript

File: src/js-v2-api-specification.md [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+##### Parameters
+
+| Parameters              |                                                                                                                                                                |
+|-------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| overlayUri                 | **Type:** Uri *Required*<br>The page to open in the browser overlay.                                                                                                     |
+
+##### Example Usage
+
+```js
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md]: Consistency with above: ```javascript

File: src/js-v2-api-specification.md [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+
+##### Description
+
+Redirects the user to a specified page that will open in a browser overlay view.
+
+##### Parameters
+
+| Parameters              |                                                                                                                                                                |
+|-------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| overlayUri                 | **Type:** Uri *Required*<br>The page to open in the browser overlay.                                                                                                     |
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md]: Not sure how much it matters, but whitespace alignment.

File: src/js-v2-api-specification.md [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+`nhsapp.navigation.openExternalBrowser(browserUri: Uri)`
+
+##### Description
+
+Redirects the user to a specified page that will open in the device's default browser application.
+
+##### Parameters
+
+| Parameters              |                                                                                                                                                                |
+|-------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md:209]: Not sure how much it matters, but whitespace alignment.

File: src/js-v2-api-specification.md [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+| browserUri              | **Type:** Uri *Required*<br>The page to open in the browser.                                                                                                     |
+
+##### Example Usage
+
+```js
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md]: Consistency with above: ```javascript

File: src/js-v2-api-specification.md [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+
+Live
+
+---
+
+### Storage <a name="storage"></a>
+
+#### addEventToCalendar Function <a name="addToCalendar"></a>
+
+```	javascript
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md:228]: Whitespace between ``` and javascript

File: src/js-v2-api-specification.md [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+)
+```
+
+##### Description
+
+Add an event to the calendar on the user's device. The NHS App's native controls will alert the user of any issues adding the event to their calendar.
+
+#### Parameters
+
+| Parameters              |                                                                                                                                                                |
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md:244]: Not sure how much it matters, but whitespace alignment.

File: src/js-v2-api-specification.md [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+| subject                 | **Type:** string *Required*<br>The Subject line for the calendar appointment.                                                                                                            |
+| body                    | **Type:** string *Required*<br>Plain text body for the appointment.                                                                                                                      |
+| location                | **Type:** string *Required*<br>Plain text location for the appointment.                                                                                                                  |
+| startTimeEpochInSeconds | **Type:** number *Required*<br>Start datetime for the appointment, set as a Epoch timestamp.                                                                                             |
+| endTimeEpochInSeconds   | **Type:** number *Optional*<br>End datetime for the appointment, set as a Epoch timestamp. Must be greater or equal to `startTimeEpochInSeconds`.  Defaults to the same value as the start time (matching default behaviour of iOS and Android). |
+
+##### Example Usage
+
+```js
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md]: Consistency with above: ```javascript

File: src/js-v2-api-specification.md [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+
+##### Status
+
+Live
+
+---
+
+#### downloadFromBytes Function <a name="downloadFromBytes"></a>
+
+```	javascript
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md]: Whitespace between ``` and javascript.

File: src/js-v2-api-specification.md [hunk 12]
```diff
@@ -0,0 +1,309 @@
...
+
+Download a file to the user's device. The NHS App's native controls will prompt the user to select a location.
+
+The mime type `application/vnd.apple.pkpass` is supported on the iOS platform and can be used to add a pass into the iOS wallet. The [getAppPlatform](/nhsapp-developer-documentation/js-api-specification#getAppPlatform) api can be used to determine the current application platform.
+
+#### Parameters
+
+| Parameters              |                                                                                                                                                                |
+|-------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| base64data                 | **Type:** string *Required*<br>A base64 string representation of the file.                                                                                                            |
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md]: Not sure how much it matters, but whitespace alignment.

File: src/js-v2-api-specification.md [hunk 13]
```diff
@@ -0,0 +1,309 @@
...
+| filename                    | **Type:** string *Required*<br>The name to download the file, including extension.                                                                                                                      |
+| mimeType                | **Type:** string *Required*<br>The mime type of the file.                                                                                                                  |
+
+##### Example Usage
+
+```js
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md]: Consistency with above: ```javascript

File: src/js-v2-api-specification.md [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+
+Returns a string indicating the platform on which the current page is being displayed. Check is based on the current User Agent.
+
+The returned string will be one of the following:
+
+- "android" - indicates that the page is being displayed within the native Android NHS App.
+- "ios" - indicates that the page is being displayed within the native iOS NHS App.
+- "none" - indicates that the page is not being displayed within the native NHS App (i.e. it is probably being displayed in a web browser).
+
+##### Status
```
Comment (ThomasPedleyNHS) [inline src/js-v2-api-specification.md:55]: Are any of these Status sections valuable if they're all "Live"?

**Review by ThomasPedleyNHS — CHANGES_REQUESTED** [review]
The content looks good. I've left some comments, mostly about formatting.

File: src/js-v2-api-specification.md (hunk 1 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md]: js is replaced with javascript

File: src/js-v2-api-specification.md (hunk 2 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md]: js is replaced with javascript

File: src/js-v2-api-specification.md (hunk 4 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md]: js is replaced with javascript

File: src/js-v2-api-specification.md (hunk 5 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md]: fixed

File: src/js-v2-api-specification.md (hunk 11 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md]: white spce is required for javascripts for displaying the right color of its contents. So I have added whitespace everywhere before javascript text.

File: src/js-v2-api-specification.md (hunk 14 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md:55]: This is existing text. Might be having some different purpose.That's reason I am not deleting.

File: src/js-v2-api-specification.md (hunk 13 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md]: js is replaced with javascript

File: src/js-v2-api-specification.md (hunk 12 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md]: fixed

File: src/js-v2-api-specification.md (hunk 3 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md]: js is replaced with javascript

File: src/js-v2-api-specification.md (hunk 6 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md:209]: fixed

File: src/js-v2-api-specification.md (hunk 8 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md:228]: whitespace is required before javascript text

File: src/js-v2-api-specification.md (hunk 9 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md:244]: fixed

File: src/js-v2-api-specification.md (hunk 7 above)
Comment (sanjaykumarNHS24) [reply src/js-v2-api-specification.md]: js is replaced with javascript

**Review by ThomasPedleyNHS — APPROVED** [review]

**Pull request merged**